src/network/message.rs

```diff
@@ -37,21 +37,29 @@ use consensus::encode::MAX_VEC_SIZE;
 #[derive(PartialEq, Eq, Clone, Debug)]
 pub struct CommandString(Cow<'static, str>);
 
-impl fmt::Display for CommandString {
-    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
-        f.write_str(self.0.as_ref())
-    }
-}
-
-impl From<&'static str> for CommandString {
-    fn from(f: &'static str) -> Self {
-        CommandString(f.into())
+impl CommandString {
+    /// Convert from various string types into a [CommandString].
+    ///
+    /// Supported types are:
+    /// - `&'static str`
+    /// - `String`
+    /// - `&str`
+    ///
+    /// Returns an empty error if and only if the string is
+    /// larger than 12 characters in length.
+    pub fn from<S: Into<Cow<'static, str>>>(s: S) -> Result<CommandString, ()> {
```

Comment:
nit, Calling the function `from` outside the `From` trait is somewhat weird

Reply:
Yeah I couldn't think of a better name. I thought of `create` but it's quite conventional to use `from` or `from_xxx` for converter methods. I also considered just `new`. Open to opinions, though, don't feel strongly about this. I just thought from was good enough.

Reply:
> Open to opinions

Personally, `new` would probably be the first thing I would look for when trying to construct one of these.

Another thing to consider might be to name it `try_from` given that you are returning a `Result`.

Reply:
According to rust API guidelines we should use `with_str` as a function name: https://rust-lang.github.io/api-guidelines/naming.html#casing-conforms-to-rfc-430-c-case

Reply:
I'm not sure, the guildlines say `with_more_details`, however in this case intput is not "more details", it's literally the source of informtion. My understanding is that `with_more_details` is used for special cases (`with_capacity` is an obvious one)

The function header also kinda leaks internal representation but I guess it's fine?
Implementing `TryFrom` would be best, but AFAIR it's not in supported MSRV.